core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -42,34 +48,50 @@ impl TeeRequestProcessor {
         let mut missing_range: Option<(L1BatchNumber, L1BatchNumber)> = None;
 
         let result = loop {
-            let l1_batch_number = match self
+            let locked_batch = match self
                 .lock_batch_for_proving(request.tee_type, min_batch_number)
                 .await?
             {
                 Some(number) => number,
                 None => break Ok(Json(TeeProofGenerationDataResponse(None))),
             };
+            let batch_number = locked_batch.l1_batch_number;
 
-            match self.blob_store.get(l1_batch_number).await {
-                Ok(input) => break Ok(Json(TeeProofGenerationDataResponse(Some(Box::new(input))))),
-                Err(ObjectStoreError::KeyNotFound(_)) => {
+            match self.blob_store.get_optional(batch_number).await {
+                Ok(Some(input)) => {
+                    break Ok(Json(TeeProofGenerationDataResponse(Some(Box::new(input)))))
+                }
+                Ok(None) => {
                     missing_range = match missing_range {
-                        Some((start, _)) => Some((start, l1_batch_number)),
-                        None => Some((l1_batch_number, l1_batch_number)),
+                        Some((start, _)) => Some((start, batch_number)),
+                        None => Some((batch_number, batch_number)),
+                    };
+                    let datetime_utc = Utc.from_utc_datetime(&locked_batch.created_at);
+                    let duration = Utc::now().signed_duration_since(datetime_utc);
+                    let status = if duration > ChronoDuration::days(10) {
+                        TeeProofGenerationJobStatus::PermanentlyIgnored
```

Comment:
Nit: Maybe it makes sense to log this status assignment, so that it is easier to debug?

Reply:
Good point. Added. ✅